test/simulator/timestamp/src/connection_simulator.cpp

```diff
@@ -94,45 +95,49 @@ connection_simulator::query_timestamp(const std::string &config, std::string &he
 
         /* Throw an error if the config cannot be parsed. */
         WT_SIM_RET_MSG(ts_manager->parse_config(config, config_map),
-            "Incorrect config passed to query timestamp: " + config);
+          "Incorrect config passed to query timestamp: " + config);
 
         auto pos = config_map.find("get");
-        if (pos != config_map.end()) {
-            query_timestamp = pos->second;
-            config_map.erase(pos);
-        } else
-            throw std::runtime_error("Incorrect config (" + config + ") passed in query timestamp");
+        if (pos == config_map.end())
+            WT_SIM_RET_MSG(EINVAL, "Incorrect config (" + config + ") passed in query timestamp");
+
+        query_timestamp = pos->second;
+        config_map.erase(pos);
 
         if (!config_map.empty())
```

Comment:
Instead of checking for empty, since we call `erase` just before, can we just check the length is 1? That would save an `erase` call.

Reply:
I am confused again about why we are expecting an empty map here. If we expect a unique entry, maybe we can do better than using a map?

Reply:
The function parse_config is not limited to just this function. It will be used in all the functions that need config parsing. Hence, I want to keep the config parsing method consistent everywhere, that is, I want to use the same function wherever config parsing is required for consistency.

We expect an empty map here because if the config is "get=oldest,blah", we want to error out on this config. 

I agree with your first comment, we could just check the size of the map.

Reply:
Updated in [c8ab007](https://github.com/wiredtiger/wiredtiger/pull/8167/commits/c8ab007f912b9ae0c9e0c62956b84df8fb4a49ba)